src/components/Button/index.tsx

```diff
@@ -1,36 +1,45 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
-import { Tooltip } from './Tooltip'
-import './styles/button.less'
+
+import { Tooltip } from '../Tooltip'
+
+import underscore from '../../utils/underscore'
+
+import styles from './Button.module.less'
 
 type NativeButtonProps = React.DetailedHTMLProps<
     React.ButtonHTMLAttributes<HTMLButtonElement>,
     HTMLButtonElement
 >
 
-export type ButtonProps = Omit<NativeButtonProps, 'title' | 'ref'> & {
-    /**
-     * Loading style. When true it disables the button, but it also adds a visual indication of
-     * some in-progress operation going on.
-     */
-    loading?: boolean
-    /**
-     * Controls visually how the button shows up from a predefined set of kinds of buttons.
-     */
-    variant?: 'primary' | 'secondary' | 'danger' | 'link'
-    /**
-     * The size of the button. If not provided it shows up in a normal size.
-     */
-    size?: 'default' | 'small' | 'large'
-    /**
-     * Tooltip that is displayed on hover.
-     *
-     * This replaces `title` which is not supported for these buttons to avoid confusion.
-     */
-    tooltip?: React.ReactNode
+type ThemeProps = {
+    theme?: 'twistLight' | 'twistDark'
```

Comment:
I know this PR is an experiment, so this is probably temporary. But it strikes as odd that there's a theme definition inside the button module. A theme should be something more global. Also the names having "twist" seems odd.